### ISSUE TITLE
deftest without docstring silently drops first step

### ISSUE THREAD

**Issue by aengelberg** [amperity/greenlight#13]
The following test:
```clojure
(deftest my-test
  #::step{:name 'step-1}
  #::step{:name 'step-2})
```
will only execute step 2 because it thinks step 1 is the docstring.

I'd prefer docstrings to be optional in `deftest`s, or for `deftest` to throw an error if the second argument isn't a string.

**Comment by trevorbernard** [comment]
> Define a new integration test. The test attributes may be a simple string description or a map of configuration to merge into the test. Defines a function which will construct the test config map. Steps can be individual steps or an arbitrarily nested sequential collection of steps.

I made the docstring optional but I changed the semantics of deftest based on your comments. The original deftest treated test-attributes as either a docstring or a configuration map, followed by a sequence of steps. If you want to maintain the original semantics, I can walk the first element after the optional docstring to see if it contains :greenlight.step/keys or treat it is a config map.

**Comment by jstokes** [comment]
Closed by #31
